sotodlib/sim_hardware.py

```diff
@@ -686,8 +741,18 @@ def sim_telescope_detectors(hw, tele, tube_slots=None):
                 alldets.update(dets)
                 windx += 1
             tindx += 1
-    return alldets
 
+    # Transform quats from left-handed layout coordinates to
+    # right-handed TOAST-compatible coordintes -- see note at top of
+    # this file.  Note the ToastishQuat is just helping us convert to
+    # and from 3g quats.
+    quats_in = ToastishQuat([d['quat'] for d in alldets.values()])  # shape (n, 4)
+    xi, eta, gamma = so3q.decompose_xieta(quats_in.to_g3())
+    quats_out = ToastishQuat(so3q.rotation_xieta(eta, xi, np.pi/2 - gamma))
```

Comment:
This is a reasonable stop-gap solution, but keep in mind we are re-writing the contents of this file to do the layout and projection separately.

Reply:
The code is now correct and accurately documented; but I will not object to replacement with a superior thing that is also correct and accurately documented.